internal/internal_task_pollers.go

```diff
@@ -227,26 +221,77 @@ func (wtp *workflowTaskPoller) processWorkflowTask(workflowTask *workflowTask) e
 		return nil
 	}
 
-	startTime := time.Now()
-	// Process the task.
-	completedRequest, wc, err := wtp.taskHandler.ProcessWorkflowTask(workflowTask.task, workflowTask.historyIterator)
-	if err == nil && completedRequest == nil {
-		if wc != nil {
-			// decision task cannot complete, we need a timer
-			wtp.scheduleRespondDecisionTaskCompleted(wc, workflowTask, startTime)
+	doneCh := make(chan struct{})
+	laResultCh := make(chan *localActivityResult)
+	// close doneCh so local activity worker won't get blocked forever when trying to send back result to laResultCh.
+	defer close(doneCh)
+
+process_WorkflowTask_Loop:
+	for {
+		startTime := time.Now()
+		workflowTask.doneCh = doneCh
+		workflowTask.laResultCh = laResultCh
+		completedRequest, wc, err := wtp.taskHandler.ProcessWorkflowTask(workflowTask)
+		if err == nil && completedRequest == nil {
+			// decision task cannot complete, we need a timer to force complete it
+		wait_LocalActivity_Loop:
+			for {
+				deadlineToTrigger := time.Duration(float32(0.8) * float32(wc.GetDecisionTimeout()))
```

Comment:
Where does this 0.8 coming from? May be make it a constant with self describing name?

Reply:
added.